### ISSUE TITLE
Crash when delete first measure

### ISSUE THREAD

**Issue by DmitryArefiev** [musescore/MuseScore#17726]
### Issue type

Crash or freeze

### Bug description

Can't reproduce with each of my scores but some of them

### Steps to reproduce

1. Open [1.zip](https://github.com/musescore/MuseScore/files/11564643/1.zip)
2. Delete 1st measure (with Ctr/Cmd)

### Screenshots/Screen recordings

https://github.com/musescore/MuseScore/assets/90187801/8d812236-399e-44a5-9bac-8b6bb552dd17

### MuseScore Version

4.1

### Regression

Yes, this used to work in **MuseScore 4.0.2**

### Operating system

Windows10

**Comment by Jojo-Schmitz** [comment]
Stach trace:
```
>	MuseScore4.exe!std::_Tree<std::_Tmap_traits<int,mu::engraving::TimeSig *,std::less<int>,std::allocator<std::pair<int const ,mu::engraving::TimeSig *>>,0>>::_Find_lower_bound<int>(const int & _Keyval) Line 1611	C++
 	MuseScore4.exe!std::map<int,mu::engraving::TimeSig *,std::less<int>,std::allocator<std::pair<int const ,mu::engraving::TimeSig *>>>::_Try_emplace<int>(int && _Keyval) Line 197	C++
 	MuseScore4.exe!std::map<int,mu::engraving::TimeSig *,std::less<int>,std::allocator<std::pair<int const ,mu::engraving::TimeSig *>>>::operator[](int && _Keyval) Line 174	C++
 	MuseScore4.exe!mu::engraving::Staff::addTimeSig(mu::engraving::TimeSig * timesig) Line 661	C++
 	MuseScore4.exe!mu::engraving::Segment::add(mu::engraving::EngravingItem * el) Line 652	C++
 	MuseScore4.exe!mu::engraving::Score::addElement(mu::engraving::EngravingItem * element) Line 1578	C++
 	MuseScore4.exe!mu::engraving::AddElement::redo(mu::engraving::EditData * __formal) Line 881	C++
 	MuseScore4.exe!mu::engraving::UndoStack::push(mu::engraving::UndoCommand * cmd, mu::engraving::EditData * ed) Line 357	C++
 	MuseScore4.exe!mu::engraving::Score::undo(mu::engraving::UndoCommand * cmd, mu::engraving::EditData * ed) Line 4361	C++
 	MuseScore4.exe!mu::engraving::Score::undoAddElement(mu::engraving::EngravingItem * element, bool addToLinkedStaves, bool ctrlModifier) Line 5778	C++
 	MuseScore4.exe!mu::engraving::Score::deleteMeasures(mu::engraving::MeasureBase * mbStart, mu::engraving::MeasureBase * mbEnd, bool preserveTies) Line 2989	C++
 	MuseScore4.exe!mu::engraving::Score::doTimeDelete(mu::engraving::Segment * startSegment, mu::engraving::Segment * endSegment) Line 4382	C++
 	MuseScore4.exe!mu::engraving::Score::cmdTimeDelete() Line 4301	C++
 	MuseScore4.exe!mu::notation::NotationInteraction::removeSelectedRange() Line 4367	C++
 	MuseScore4.exe!mu::notation::NotationActionController::registerAction::__l2::<lambda>() Line 2079	C++
 	[External Code]	
 	MuseScore4.exe!mu::actions::IActionsDispatcher::reg::__l2::<lambda>(const std::string & __formal, const mu::actions::ActionData & __formal) Line 50	C++
 	[External Code]	
 	MuseScore4.exe!mu::actions::ActionsDispatcher::dispatch(const std::string & actionCode, const mu::actions::ActionData & data) Line 69	C++
 	MuseScore4.exe!mu::actions::ActionsDispatcher::dispatch(const std::string & actionCode) Line 43	C++
 	MuseScore4.exe!mu::shortcuts::ShortcutsController::activate(const std::string & sequence) Line 46	C++
 	MuseScore4.exe!mu::shortcuts::ShortcutsInstanceModel::doActivate(const QString & key) Line 83	C++
 	MuseScore4.exe!mu::shortcuts::ShortcutsInstanceModel::activate(const QString & key) Line 59	C++
 	MuseScore4.exe!mu::shortcuts::ShortcutsInstanceModel::qt_static_metacall(QObject * _o, QMetaObject::Call _c, int _id, void * * _a) Line 104	C++
 	MuseScore4.exe!mu::shortcuts::ShortcutsInstanceModel::qt_metacall(QMetaObject::Call _c, int _id, void * * _a) Line 170	C++
 	[External Code]	
 	MuseScore4.exe!mu::app::App::run(int argc, char * * argv) Line 325	C++
 	MuseScore4.exe!main(int argc, char * * argv) Line 335	C++
 	[External Code]	
```

**Comment by DmitryArefiev** [comment]
Fixed in #17733
